frontend/test/resolution/testTypePropertyPrimitives.cpp

```diff
@@ -0,0 +1,580 @@
+/*
+ * Copyright 2021-2023 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "test-resolution.h"
+
+#include "chpl/parsing/parsing-queries.h"
+#include "chpl/resolution/resolution-queries.h"
+#include "chpl/resolution/scope-queries.h"
+#include "chpl/resolution/can-pass.h"
+#include "chpl/types/all-types.h"
+#include "chpl/uast/Identifier.h"
+#include "chpl/uast/Module.h"
+#include "chpl/uast/Record.h"
+#include "chpl/uast/Variable.h"
+
+struct Test {
+  enum Output {
+    TRUE,
+    FALSE,
+    STRING,
+    ERROR 
+  };
+
+  struct PrimitiveCalls {
+    std::vector<const char*> actuals;
+    Output expected;
+    std::string str;
+
+    PrimitiveCalls(std::vector<const char*> actuals, Output expected,
+                   std::string str="")
+      : actuals(std::move(actuals)),
+        expected(expected),
+        str(std::move(str)) {
+      if (expected != STRING) assert(this->str.empty());
+    }
+  };
+
+  std::string testName;
+  bool isChplHomeRequired = false;
+  std::string prelude;
+  chpl::uast::primtags::PrimitiveTag primitive;
+  std::vector<PrimitiveCalls> calls;
+};
+
+static bool
+isParamStringMatch(chpl::types::QualifiedType qt, std::string str,
+                   std::string* out) {
```

Comment:
`out` is never `null`, consider making it a reference parameter and removing the `if (out)` check.